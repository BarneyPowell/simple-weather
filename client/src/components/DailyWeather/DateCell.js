import React, { Component } from  'react';
import Moment from 'moment';

class DateCell extends Component {
    render() {
        const date = Moment(this.props.date);
        

        return (
            <span className="daily-weather_day">
                <span className="-date">{date.format("DD MMM")}</span>
                <span className="-day"><span>{date.format("ddd")}</span></span>
            </span>
        );
    }
}

export default DateCell;