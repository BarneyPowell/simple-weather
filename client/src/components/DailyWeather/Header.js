import React from 'react';

export default (props) => {
    return (
        <div className="daily-weather_header">
            <span className="-intro">5-day forecast for</span>
            <span className="-location">{`${props.location.name}, ${props.location.country}`}</span>
        </div>
    );
}