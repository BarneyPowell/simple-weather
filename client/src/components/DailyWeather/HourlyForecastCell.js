import React, { Component } from 'react';
import Moment from 'moment';
import PropTypes from 'prop-types';

class HourlyForecastCell extends Component {
    constructor(props) {
        super(props);
    }

    render() {

        const { temperature } = this.props;

        let displayTemperature = Math.round(temperature);

        const time = Moment(this.props.time);

        const format = this.props.is24h
            ? "HH:mm"
            : "ha";
            
        return (
            <li className="hourly-forecast">
                <span className="-hour">{time.format(format)}</span>
                <span className="-temperature"><span>{displayTemperature}&deg;</span></span>
               
            </li>
        )
    }
}

HourlyForecastCell.propTypes = {
    temperature: PropTypes.number,
    hour: PropTypes.number,
    is24h: PropTypes.bool
}


export default HourlyForecastCell;