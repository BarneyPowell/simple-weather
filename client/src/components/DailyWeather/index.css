


.daily-weather ol {
    list-style: none;
    margin: 0;
    padding: 0;
}


.daily-weather_days {
    display: flex;
    flex-direction: column;
}

.daily-weather_days > li {
    display: flex;
    flex-direction: row;
    flex: 0 1 auto;
    border-bottom: 1px solid #abcdef;
}


.daily-weather_day {
    width: 10%;
    flex: 0 1 auto;

    background-color: #ddeeff;

    height: 4em;
    width: 4em;

    display: flex;
    flex-direction: column;
}

.daily-weather_day .-date {
    font-size: 0.625em;

    flex: 0 1 auto;
    text-align: center;
    padding: 0.2em;
}

.daily-weather_day .-day {
    flex: 0 1 auto;
    height: 100%;
    font-size: 1.4em;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
}

.daily-weather_day .-day span {
    flex: 0 1 auto;
}

.daily-weather_hours {
    display: flex;
    width: 100%;
    flex: 0 1 auto;
    overflow: auto;
}


.hourly-forecast {
    flex: 0 0 auto;
    width: 4em;
    height: 4em;
    position: relative;
    display: flex;
    flex-direction: column;
}


.hourly-forecast .-temperature {
    flex: 0 1 auto;
    display: flex;
    height: 100%;
    align-items: center;
    justify-content: center;
}

.hourly-forecast .-temperature span {
    flex: 0 1 auto;
}

.hourly-forecast .-hour {
    flex: 0 1 auto;
    text-align: center;
    font-size: 0.625em;
    padding: 0.2em;
}


