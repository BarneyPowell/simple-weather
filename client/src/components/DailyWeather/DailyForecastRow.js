import React from 'react';

import DateCell from './DateCell';
import HourlyForecastCell from './HourlyForecastCell';

export default (props) => {
    const { day } = props;
    return (
        <li >
        <DateCell date={day.day} />
        <ol className="daily-weather_hours">
            {day.hours.map((hourly, idx) => 
                <HourlyForecastCell key={`${hourly.time}`} 
                    temperature={hourly.temp} 
                    time={hourly.time}
                    weather={hourly.weather} />
            )}
        </ol>
    </li>
    );
}