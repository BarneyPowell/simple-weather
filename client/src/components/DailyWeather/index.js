import React, { Component } from 'react';
import PropTypes from 'prop-types';

import HourlyForecastCell from './HourlyForecastCell';
import DateCell from './DateCell';

import './index.css';

import generateTestWeather from '../../lib/generateTestWeather';




class DailyWeather extends Component
{

    constructor(props) {
        super(props);


    }

    render() {

        const classNames = [
            this.props.className
        ];

        classNames.push('daily-weather');

        const forecast = this.props.forecast || [];
        

        return (
            <div className={classNames.join(' ')}>
                <ol className="daily-weather_days">
                    {forecast.map((day, idx) => 
                        <li key={`day.date`}>
                            <DateCell date={day.date} />
                            <ol className="daily-weather_hours">
                                {day.hours.map((hourly, idx) => 
                                    <HourlyForecastCell key={`${day.date.toString()}_${idx}`} temperature={hourly.temp} time={hourly.time} />
                                )}
                            </ol>
                        </li>
                    )}
                </ol>
            </div>
        );
    }
}

DailyWeather.propTypes = {
    location: PropTypes.object,
    forecast: PropTypes.array
}

export default DailyWeather;