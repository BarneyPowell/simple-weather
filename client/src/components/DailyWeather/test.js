import React from 'react';
import DailyWeather from './';

it('should render without crashing', () => {
    const wrapper = shallow(<DailyWeather />);
    expect(wrapper).toMatchSnapshot();
});


it('should render with the supplied classname', () => {
    const wrapper = shallow(<DailyWeather className='test-weather' />);
    expect(wrapper.props().className).toContain('test-weather');
});