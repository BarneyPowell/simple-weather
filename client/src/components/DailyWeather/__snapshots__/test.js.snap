// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className=" daily-weather"
>
  <ol
    className="daily-weather_days"
  />
</div>
`;
